server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -554,4 +563,55 @@ public DruidDataSource apply(DruidServer input)
     segments.put("maxTime", new DateTime(maxTime));
     return retVal;
   }
+
+  /**
+   * Provides serverView for a datasource and Interval which gives details about servers hosting segments for an interval
+   * Used by the realtime tasks to fetch a view of the interval they are interested in.
+   */
+  @GET
+  @Path("/{dataSourceName}/intervals/{interval}/serverview")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getSegmentDataSourceSpecificInterval(
+      @PathParam("dataSourceName") String dataSourceName,
+      @PathParam("interval") String interval,
+      @QueryParam("partial") final boolean partial
+  )
+  {
+    TimelineLookup<String, SegmentLoadInfo> timeline = serverInventoryView.getTimeline(
+        new TableDataSource(dataSourceName)
+    );
+    final Interval theInterval = new Interval(interval.replace("_", "/"));
+    if (timeline == null) {
+      log.debug("No timeline found for datasource[%s]", dataSourceName);
+      return Response.noContent().build();
```

Comment:
Why not return an empty array? It's ok for a datasource/interval pair to not have any served segments.